### PR TITLE
Integrationtesting

### PR DESCRIPTION
**Pull request by clunaslunas**
Fix issue #31

### CODE REVIEW

File: cmd/get.go [hunk 1]
```diff
@@ -64,18 +59,14 @@ to decrypt it securely.`,
...
-			Client:          svc.S3Client,
+		getobj := &GetObject{
 			Bucket:          s3bucket,
 			Key:             s3object,
 			FileDestination: filedest,
 		}
+		decryptionclient := NewDecryptionClient()
 
-		content, err := decryptionclient.getS3Cse()
+		content, err := GetS3Cse(getobj, decryptionclient)
```
Comment (danielc2013) [inline cmd/get.go]: So I'm not sure how I feel about this implementation:
- What's the benefit of removing composition? Seems like that would be more cohesive, and maybe more portable. Unless this is necessary for how your tests are setup?
 i.e. `func (g *GetObject) GetS3Cs3(decryptionclient S3DecryptionClientAPI) (io.ReadCloser, error) {}`
- Also, you're handling the error in GetS3Cse and then returning it but not actually utilizing it after you return it. So maybe just don't return the error, or wait to handle it after the call?

File: cmd/put.go [hunk 2]
```diff
@@ -98,20 +87,21 @@ a file to S3.`,
...
 	Bucket   string
 	Key      string
 	Source   string
 	Reader   *bufio.Reader
 	ByteSize int
 }
 
-func (p *Put) putS3Cse() ([]byte, error) {
+//PutS3Cse puts encrypted objects into S3
+func PutS3Cse(p *PutObject, encryptionclient S3ClientPutAPI) ([]byte, error) {
```
Comment (danielc2013) [inline cmd/put.go:100]: Same question here about composition

**Comment by nz285** [comment]
I compiled and tested it.  It does its job.  A couple of observations:
- The `get object by version` code is missing in `get`.  Will need to be added back in.  I believe they are in the `staging` branch.
- It would be good to update `readme` to reflect how to run tests locally.

File: cmd/get.go (hunk 1 above)
Comment (clunaslunas) [reply cmd/get.go]: Resolving this conversation with 8835e77 change

File: cmd/put.go (hunk 2 above)
Comment (clunaslunas) [reply cmd/put.go:100]: Implementing interfaces correctly to set up unit tests will occur first, then we can build up our composition.

**Comment by nz285** [comment]
👍

**Review by danielc2013 — APPROVED** [review]
Thanks Carlos!

**Pull request merged**